src/components/fields/ArrayField.js

```diff
@@ -379,14 +376,7 @@ class ArrayField extends Component {
         );
       }
 
-      if (isDefinedFormWithNonEmptyProperties) {
-        schema = {
-          ...schema,
-          items: schema.items[0],
-        };
-
-        delete schema.additionalItems;
-      }
+      // let renderFixedArray handle forms with non-empty properties
```

Comment:
previously, I modified it so it became a normal array schema. That path does not pre-populate array items. I changed it to just leave it as a fixed array schema, and then add custom logic in renderFixedArray to handle forms specifically -- other code remains the same.